Resources/Storage/GFAL2_StorageBase.py

```diff
@@ -289,7 +289,7 @@ def __putSingleFile( self, src_file, dest_url, sourceSize ):
           errStr = "GFAL2_StorageBase.__putFile: The local source file does not exist or is a directory"
           self.log.error( errStr, src_file )
           return S_ERROR( errStr )
-        src_url = 'file:%s' % src_file
+        src_url = 'file://%s' % os.path.abspath( src_file )
```

Comment:
Btw, I just realized that depending where we get the source from, it might already contains 'file://' at the beginning (for example, if it is SRM giving it to us). Please do the test (everywhere obviously) before adding it